src/main/java/bio/terra/service/dataset/flight/ingest/IngestPopulateFileStateFromFlightMapStep.java

```diff
@@ -51,7 +56,16 @@ public StepResult doSkippableStep(FlightContext context) {
     List<String> errors = new ArrayList<>();
     try (var bulkFileLoadModels = getModelsStream(ingestRequest, fileColumns, errors)) {
 
-      loadService.populateFiles(loadId, bulkFileLoadModels, batchSize);
+      List<FileModel> existingFiles = new ArrayList<>();
+
+      if (ingestRequest.isResolveExistingFiles()) {
+        var toIngest =
+            bulkFileLoadModels.flatMap(fileModel -> fileNotIngested(fileModel, existingFiles));
```

Comment:
It looks like what you want here is a way to split the stream into two different collections; I think `Collectors.partitioningBy()` can do this.

Reply:
`Collectors.partitionBy` would be a terminal operation on the stream, which we don't want to happen until the results are being realized in batches when being inserted into the File Load table. This side-effecting `flatMap` collects the files that exist as the batches are realized so that the entire stream isn't loaded into memory.